.github/workflows/e2ePerformanceTests.yml

```diff
@@ -4,117 +4,139 @@ on:
         env:
           GITHUB_TOKEN: ${{ github.token }}
 
-      - name: Get pull request details
-        id: getPullRequestDetails
-        uses: Expensify/App/.github/actions/javascript/getPullRequestDetails@main
-        with:
-          GITHUB_TOKEN: ${{ github.token }}
-          PULL_REQUEST_NUMBER: ${{ inputs.PR_NUMBER }}
-          user: ${{ github.actor }}
-
-      - name: Get merge commit sha for the pull request
-        id: getMergeCommitSha
-        run: echo "MERGE_COMMIT_SHA=${{ steps.getPullRequestDetails.outputs.MERGE_COMMIT_SHA }}" >> "$GITHUB_OUTPUT"
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
-
-  buildBaseline:
-    runs-on: ubuntu-20.04-64core
-    needs: [prepare]
-    name: Build apk from latest release as a baseline
-    steps:
-      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
-        with:
-          fetch-depth: 0
-
       - name: Check if there's an existing artifact for this baseline
         id: checkForExistingArtifact
         uses: xSAVIKx/artifact-exists-action@3c5206b1411c0d2fc0840f56b7140646933d9d6a
         with:
-          name: baseline-apk-${{ needs.prepare.outputs.VERSION }}
+          name: baseline-apk-${{ steps.getMostRecentRelease.outputs.VERSION }}
 
       - name: Skip build if there's already an existing artifact for the baseline
         if: ${{ fromJSON(steps.checkForExistingArtifact.outputs.exists) }}
-        run: echo 'APK for baseline ${{ needs.prepare.outputs.VERSION }} already exists, reusing existing build'
+        run: echo 'APK for baseline ${{ steps.getMostRecentRelease.outputs.VERSION }} already exists, reusing existing build'
 
       - name: Checkout "Baseline" commit (last release)
         if: ${{ !fromJSON(steps.checkForExistingArtifact.outputs.exists) }}
-        run: git checkout ${{ needs.prepare.outputs.VERSION }}
+        run: git checkout ${{ steps.getMostRecentRelease.outputs.VERSION }}
 
       - name: Build APK
         if: ${{ !fromJSON(steps.checkForExistingArtifact.outputs.exists) }}
         uses: Expensify/App/.github/actions/composite/buildAndroidAPK@main
         with:
-          ARTIFACT_NAME: baseline-apk-${{ needs.prepare.outputs.VERSION }}
+          ARTIFACT_NAME: baseline-apk-${{ steps.getMostRecentRelease.outputs.VERSION }}
 
   buildDelta:
-    runs-on: ubuntu-20.04-64core
-    needs: [prepare]
+    runs-on: ubuntu-latest
     name: Build apk from delta ref
+    outputs:
+      DELTA_REF: ${{ steps.setDeltaRef.outputs.DELTA_REF }}
     steps:
       - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
         with:
           fetch-depth: 0
 
-      - name: Checkout "delta ref"
-        run: git checkout ${{ needs.prepare.outputs.DELTA_REF }}
+      - name: Get pull request details
+        id: getPullRequestDetails
+        uses: jczekalski/App/.github/actions/javascript/getPullRequestDetails@fix/15605-run-e2e-performance-tests-for-unmerged-PRs
```

Comment:
Modified for testing purposes. Change to `uses: Expensify/App/.github/actions/javascript/getPullRequestDetails@main` before merging to `main`.